ghost/core/core/server/data/migrations/versions/5.15/2022-09-14-12-46-add-email-track-clicks-setting.js

```diff
@@ -0,0 +1,59 @@
+const logging = require('@tryghost/logging');
+const ObjectId = require('bson-objectid').default;
+const {createTransactionalMigration} = require('../../utils');
+
+const {MIGRATION_USER} = require('../../utils/constants');
+const key = 'email_track_clicks';
+const cloneValueOf = 'email_track_opens';
+const defaultValue = 'true';
+const group = 'email';
+const type = 'boolean';
+
+// Note: we cannot use the addSetting helper here because we need to base the value on the current value of email_track_opens
+module.exports = createTransactionalMigration(
+    async function up(connection) {
+        const settingExists = await connection('settings')
+            .where('key', '=', key)
+            .first();
+        if (settingExists) {
+            logging.warn(`Skipping adding setting: ${key} - setting already exists`);
+            return;
+        }
+
+        // Value is based on the email_track_opens value
+        const reuseValueOfSetting = await connection('settings')
+            .where('key', '=', cloneValueOf)
+            .first();
+
+        logging.info(`Adding setting: ${key}`);
+        const now = connection.raw('CURRENT_TIMESTAMP');
+
+        return connection('settings')
+            .insert({
+                id: ObjectId().toHexString(),
+                key,
+                value: reuseValueOfSetting ? reuseValueOfSetting.value : defaultValue,
+                group,
+                type,
+                created_at: now,
+                created_by: MIGRATION_USER,
+                updated_at: now,
```

Comment:
We're not very strict with doing this (yet) but can I suggest we don't add `updated_at`, `updated_by` when we insert a new entry? Strictly speaking, I'd say it's been created but not _updated_ yet

Reply:
Is it okay if I change this in the addSetting helper too:
https://github.com/SimonBackx/Ghost/blob/e259dd11b6758272c89410e586eafdb98d198f22/ghost/core/core/server/data/migrations/utils/settings.js#L34-L37

Reply:
Fine with me 👍🏻